### ISSUE TITLE
Potential Tensorflow 2.0 bug

### ISSUE THREAD

**Issue by a-jahani** [ialhashim/DenseDepth#49]
Hi,

Thank you for sharing your code. I think there might be a small bug in your tensorflow 2.0 dataloader where you shuffle in a wrong place. The way you shuffle makes all epochs the same permutation. I think you need to have a shuffle before/after repeat. 

PS . The shuffle in model.fit wont have any effects as explained in the keras documentation [here](https://keras.io/models/sequential/)
  
Suggested code:

```
 def get_batched_dataset(self, batch_size):
        self.dataset = tf.data.Dataset.from_tensor_slices((self.filenames, self.labels))
        self.dataset = self.dataset.shuffle(shuffle_buffer_size=len(self.filenames))
        self.dataset = self.dataset.repeat()
        self.dataset = self.dataset.map(map_func=self._parse_function, num_parallel_calls=tf.data.experimental.AUTOTUNE)
        self.dataset = self.dataset.batch(batch_size=batch_size)
        return dataset

```

I was also wondering if pytorch code and tensorflow 2.0 give the same result as keras. Do you have any evaluation on those two codes compared with the original keras?

Thank you,
Ali

**Comment by ialhashim** [comment]
Hi Ali!

Thanks for pointing this bug out. Did you manage to fix the bug? I would greatly appreciate a pull request for the fix.

I have not fully tested these 'experimental' ports. I was hoping to provide a start for others to make them stronger since my main concern in this research is always performance I stayed with what gave me best results. I believe a modification on TF 2 should in theory reach the same the performance since Keras uses TF. 

For PyTorch I seem to have more trouble getting the performance close to Keras, I hope someday someone can find the different problems that might cause that.
